### PR TITLE
Separate mesh and proxy config

### PR DESCRIPTION
**Pull request by kyessenov**
Complementary PR to https://github.com/istio/pilot/pull/1205#issuecomment-328166561

**Release note**:

```release-note
none
```

### CODE REVIEW

**Comment by kyessenov** [comment]
@yutongz @sebastienvas it takes more than 20 minutes to fetch the logs. Can we post the error before fetching the logs?
Also, the actual error is consumed by the test framework, so I don't expect the error in the logs...

**Comment by yutongz** [comment]
1. Yes, you are right, we have a issue to change the way we fetch logs. https://github.com/istio/test-infra/issues/469
2. Your pr (commit: c2e31fb088ab1a7865a1a85d3237aae7d6d31ba4) failed on my environement with the same error as it showed up in CI https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/709/e2e-suite-rbac-no_auth/390/, so could you reproduce this on your personal environment, run `prow/e2e-suite-rbac-no_auth.sh --skip_cleanup`
3. I don't the framework ate any error message, if you think you need more logs or error message to debug, feel free to add more logging.

**Comment by kyessenov** [comment]
Thanks for answering.
1. It'd help perception-wise to simply post error as soon as the test failed.
2/3. I'll debug.

File: install/kubernetes/istio-auth-with-cluster-ca.yaml [hunk 1]
```diff
@@ -384,10 +384,11 @@ data:
     # Uncomment the following line to enable mutual TLS between proxies
     authPolicy: MUTUAL_TLS
     mixerAddress: istio-mixer:9091
-    discoveryAddress: istio-pilot:8080
```
Comment (ldemailly) [inline install/kubernetes/istio-auth-with-cluster-ca.yaml]: shouldn't you keep that for backward compatibility for now (until we get a new pilot)

File: install/kubernetes/istio-auth-with-cluster-ca.yaml [hunk 2]
```diff
@@ -384,10 +384,11 @@ data:
...
+    defaultConfig:
+      discoveryAddress: istio-pilot:8080
+      zipkinAddress: zipkin:9411
     ingressService: istio-ingress
     statsdUdpAddress: istio-mixer:9125
-    zipkinAddress: zipkin:9411
```
Comment (ldemailly) [inline install/kubernetes/istio-auth-with-cluster-ca.yaml]: same

File: install/kubernetes/istio-auth-with-cluster-ca.yaml [hunk 3]
```diff
@@ -425,7 +426,7 @@ spec:
       serviceAccountName: istio-pilot-service-account
       containers:
       - name: discovery
-        image: gcr.io/istio-testing/pilot:4f34dbbd7e052d6f51689b301ea244588a01a73a
+        image: gcr.io/istio-testing/pilot:b133d9280cb3fd40436380de6b1a5b2631b304dd
```
Comment (ldemailly) [inline install/kubernetes/istio-auth-with-cluster-ca.yaml]: this isn't even from a PR, where is it from ?

**Pull request closed without merging**